src/NuGet.Core/NuGet.Resolver/DateTimeUtility.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Resolver
+{
+    /// <summary>
+    /// static class to provide datetime common utility apis
+    /// </summary>
+    public static class DateTimeUtility
+    {
+        /// <summary>
+        /// take timespan n return in appropriate unit like ms, or seconds, or minutes, or hours
+        /// </summary>
+        /// <param name="time">timespan</param>
+        /// <param name="type">time measuring unit</param>
+        /// <returns></returns>
+        public static double ToReadableTimeFormat(TimeSpan time, out string type)
```

Comment:
Usability-wise can this return final formatted string to avoid `out` argument? Also time units should be localized.